MATLAB/Utilities/VarianCBCT/EasyDataLoader.m

```diff
@@ -0,0 +1,51 @@
+function [proj, angles, geo] = EasyDataLoader(datafolder, varargin)
+% Load all dataset that are needed for reconstruction
+% Date: 2020-04-16
+% Author: Yi Du (yi.du@hotmail.com)
+% datafolder = 'E:\BigData\Edge\CBCT_Export\2020-01-09_144244';
+
+%% Load geometry
+[geo, ScanXML] = GeometryFromXML(datafolder);
+
+%% Motion lag correcion
+thd = 0;
+if(~isempty(varargin)&&(varargin{1}))
+    thd = str2double(ScanXML.Acquisitions.Velocity.Text)...
+        ./str2double(ScanXML.Acquisitions.FrameRate.Text);
+    thd = thd *0.95;
+end
+
+%% Load proj and angle
+[proj, angles, blk] = BatchReadXim(datafolder, thd);
+
+%% Logarithmic calculation
+proj = log(repmat(blk, [1 1 size(proj,3)])./proj);
```

Comment:
blk: blank image (or flood field image), captured with no object in field.

In simulation, we will surely meet up cases with no penetration. However, in simulation line integral is used to generate proj, while in actual x-ray imaging high penetrations will deliver low detector signals. There are no zeros in proj. That's the discrepancy between simulation and experiment.

Reply:
Ah, I see where the difference comes. In the past, with other machines/procedures I used, we would (or the machine would) first do flat field correction using light (flood) and dark (source turned off) fields and then with that result, we would then process the data with the logarithm. This means that if there is an object where there is no penetration, you would actually have zeros in the image, as you corrected for the "background noise" that detectors measure. 

In this case, it seems that you are not doing dark field correction, which means that you will never measure zero in the detector, as you say. This code makes more sense now. 

Now a side question: You are then clearly not doing dark field correction, why is that? Is it simply because you decided not to, or is it something that the Varian CBCT does not do? I have always done it in CT, even with medical data.

Reply:
I agree with you. Dark field correction should come first before any further processing. 
The only reason for not doing this is that the work is undergoing. Dark field correction, bad pixel correction, detector gain correction, and other stuff. I have to figure out how to decode these files first. 
Anyway, you are right. I will add these features later, as separate .m files. Professional users can see how each process works by analyzing the relevant .m file. General users can just use the FDK_half template for FDK recon, or EasyLoader as the footstone for iterative recon.

Reply:
Fantastic! Thanks for letting me know. Most of this is great work, sorry for the constant questions, it is just for me to better understand and ensure I dont remove important work :)